src/main/java/org/snomed/snowstorm/core/data/services/servicehook/CommitServiceHookClient.java

```diff
@@ -51,35 +65,52 @@ public void preCommitCompletion(Commit commit) throws IllegalStateException {
 			return;
 		}
 
-		final String authenticationToken = SecurityUtil.getAuthenticationToken();
 		try {
-			final HttpHeaders httpHeaders = new HttpHeaders();
-			httpHeaders.setContentType(MediaType.APPLICATION_JSON);
-			if (authenticationToken == null) {
-				logger.info("Authentication token is null.");
-			}
-			httpHeaders.add(HttpHeaders.COOKIE, authenticationToken);
-			restTemplate.postForEntity("/integration/snowstorm/commit",
+			String authenticationToken = SecurityUtil.getAuthenticationToken();
+			HttpHeaders httpHeaders = buildHttpHeaders(authenticationToken);
+			logRequest(commit, authenticationToken, commit.getBranch());
+			ResponseEntity<?> responseEntity = restTemplate.postForEntity("/integration/snowstorm/commit",
 					new HttpEntity<>(new CommitInformation(commit), httpHeaders), Void.class);
-		} catch (RestClientException e) {
-			logger.error("Commit service hook failed for branch {}, commit {}, url {}, cookie {}",
-					commit.getBranch().getPath(), commit.getTimepoint().getTime(), serviceUrl,
-					obfuscateToken(authenticationToken), e);
+			logger.info("External system returned HTTP status code {}.", responseEntity.getStatusCodeValue());
+		} catch (HttpClientErrorException.Conflict e) {
+			// External system indicates criteria has not been completed.
+			logger.error("External system indicates not all criteria have been completed.");
 			boolean promotion = commit.getCommitType().equals(Commit.CommitType.PROMOTION);
-			if (promotion && e instanceof HttpClientErrorException.Conflict) {
-				if (blockPromotion) {
-					logger.error("Promotion blocked; not all criteria have been met.");
-					throw new RuntimeServiceException("Promotion blocked; not all criteria have been met.");
-				}
+			if (promotion && this.blockPromotion) {
+				logger.info("Promotion blocked as not all criteria have been completed; throwing exception.");
+				throw new RuntimeServiceException("Promotion blocked as not all criteria have been completed.", e);
+			} else if (promotion) {
+				logger.info("Promotion will go ahead, despite not all criteria being completed, as per configuration.");
 			}
-
-			if (blockPromotion) {
-				logger.error("Promotion blocked; unexpected error.");
-				throw e;
+		} catch (ResourceAccessException e) {
```

Comment:
I think you should use a less specific exception here. Consider it's parent `RestClientException`. Otherwise we may experience unhandled exceptions.

Reply:
Thanks Kai. 

Apologies for the silly mistake. I've made that change.